reference/constraints/Hostname.rst

```diff
@@ -0,0 +1,135 @@
+Hostname
+========
+
+This constraint ensures that the given value is a valid and non-reserved
+host name (internally it uses the ``FILTER_VALIDATE_DOMAIN`` option of the
+:phpfunction:`filter_var` PHP function).
+
+.. versionadded:: 5.1
+
+    The ``Hostname`` constraint was introduced in Symfony 5.1.
+
+==========  ===================================================================
+Applies to  :ref:`property or method <validation-property-target>`
+Options     - `groups`_
+            - `message`_
+            - `payload`_
+            - `requireTld`_
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\Hostname`
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\HostnameValidator`
+==========  ===================================================================
+
+Basic Usage
+-----------
+
+To use the Hostname validator, apply it to a property on an object that
+will contain a host name.
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/Entity/Server.php
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class ServerSettings
+        {
+            /**
+             * @Assert\Hostname(message="The server name must be a valid hostname.")
+             */
+            protected $name;
```

Comment:
Would prefer $hostname here

Reply:
I tried that first ... but some examples are super confusing because the property is called `hostname` and the constraint `Hostname` so it's harder to read. That's why I used `name` for the property.